generate.go

```diff
@@ -8,35 +8,35 @@ import (
 
 )
 
-func Create(config Config, userID string) (*string, error) {
+func Create(config Config, userID string) (string, error) {
```

Comment:
Owh, didn't spot that. I'll take a look at our docs and update as needed. Might've gone a little too fast here.

Reply:
I am not sure the empty string semantic is better than the nil string in case on error. What is the go way of doing it ?

Reply:
The go way of doing things when returning an error is to return the zero value of every other returned element:
- pointer: `nil`
- string: `""`

the type of the returned element should not depend on the value we use in case of errors

Reply:
I completely agree with that. Returning pointers for the sake of being able to return nil on error looks quite convoluted here. Caller should always check the error and discard the return values if the error is set.
There are some examples in the stdlib of returned values not being the zero one on error but they tend to be quite specific and serve other needs than ours (see  e.g. https://pkg.go.dev/go/parser#ParseDir)

Here, using a pointer makes no *real* sense:
we don't need a reference to the return value after the return
there is no case where it would be useful to return a non-nil but empty string pointer

We can keep it for compatibility reasons but I strongly suggest we deprecate this and finally drop it in future versions of the SDK

Reply:
Any thoughts on that ? I'd like this PR to move forward a little.